internal/cba.go

```diff
@@ -111,11 +111,6 @@ func getTransportConfig(settings *DialSettings) (*transportConfig, error) {
 		return nil, errUniverseNotSupportedMTLS
 	}
 
-	s2aMTLSEndpoint := settings.DefaultMTLSEndpoint
-	// If there is endpoint override, honor it.
-	if settings.Endpoint != "" {
```

Comment:
Is it a breaking change for any users to remove this?

Reply:
Also, maybe the scope should be `fix` rather than `feat`? I'm not sure what feature this is adding. Seems like more of a bug fix.

Reply:
thanks
The purpose of this PR is to not use s2a (new behavior) in cases where Endpoint is set. So the library will continue to follow its default behavior, i.e. TLS to the DefaultEndpoint, which should be safe.